scipy/optimize/_linprog_util.py

```diff
@@ -1086,6 +1086,79 @@ def vstack(blocks):
     return A, b, c, c0, x0
 
 
+def _round_to_power_of_two(x):
+    """
+    Round elements of the array to the nearest power of two.
+    """
+    return 2**np.round(np.log2(x))
+
+def _autoscale(A, b, c, x0):
+    """
+    Scales the problem according to ___
+    """
+    m, n = A.shape
+
+#    with warnings.catch_warnings():
+#    warnings.simplefilter("ignore")
+    C = 1
+    R = 1
+
+    if A.size > 0:
+        if sps.issparse(A):
+            R = np.max(np.abs(A), axis=1).toarray().flatten()
+            R[R == 0] = 1
+            R = _round_to_power_of_two(R)
+            b = b/R
+            R = sps.diags(1/R)
+            A = R*A
+
+            C = np.max(np.abs(A), axis=0).toarray().flatten()
+            C[C == 0] = 1
+            C = _round_to_power_of_two(C)
+            c = c/C
+            C = sps.diags(1/C)
+            A = A*C
+
+        else:
+            R = np.max(np.abs(A), axis=1)  # problematic with sparse arrays
+            R[R == 0] = 1  # no good when array is sparse
+            R = 1/_round_to_power_of_two(R)
+            A = A*R.reshape(m, 1)
+            b = b*R
+
+            C = np.max(np.abs(A), axis=0)
+            C[C == 0] = 1
+            C = 1/_round_to_power_of_two(C)
+            A = A*C
+            c = c*C
+
+    b_scale = 1
+    if b.size > 0:
+        b_scale = np.max(np.abs(b))
```

Comment:
I'm not too fussed about this, but, I think this would be neater as a ternary operator.

```b_scale = np.max(np.abs(b)) if b.size > 0 else 1``

Reply:
Ternary or not, I should have recognized that it could have been an `else` instead of a second `if`. Implemented.